### PR TITLE
Error flow

### PR DESCRIPTION
**Pull request by yradkhorrami**
BEGINRELEASENOTES
Improved the `ErrorFlow` processor:
1.  An option added to enable/disable confusion term in the jet energy error
2. For scaling angular uncertainties of jets, CovMat of the jet is scaled by scaling factors (as processor parameters)
3. So far, charged PFOs used to be identified by charge, so the energy of neutral PFOs with tracks was added to photons/neutral hadrons energy. Now, neutral PFOs are classified as charged PFO, since the energy and momentum are obtained from tracks.
4. An option added to propagate confusion term to **all** covariance matrix elements.

- New processor parameters:
      -   "EnableConfusionTerm", "Enable/disable confusion term to be added to covariance matrix"
      -   "CovMatFactorPhotons", "A correction factor to be multiplied to angular uncertainties of photons"
      -   "CovMatFactorNeutralHadrons", "A correction factor to be multiplied to angular uncertainties of Neutral Hadrons"
      -   "PropagateConfusion2Mom", "Enable/disable Propagating uncertainty due to confusion to the Momentum components/All CovMat elements"

ENDRELEASENOTES

### CODE REVIEW

**Comment by gaede** [comment]
Please put some meaningful release notes into the commit comment, between
```
BEGINRELEASENOTES
...
ENDRELEASENOTES
```

**Comment by yradkhorrami** [comment]
@gaede , Thanks. I hope that it's clear now.

**Comment by gaede** [comment]
Thanks @yradkhorrami. I have just added a heading clarifying that this addresses the ErrorFlow proc...

**Comment by rete** [comment]
@gaede We still hve this issue with TravisCI. We need to change the CI to github actions...

**Comment by gaede** [comment]
Could you do this for MarlinReco and then the other packages as needed ?

**Comment by rete** [comment]
@yradkhorrami Can you rebase please? We have a new CI working now :-)

**Review by rete — CHANGES_REQUESTED** [review]

File: Analysis/ErrorFlow/include/ErrorFlow.h [hunk 1]
```diff
@@ -47,109 +47,114 @@ using namespace marlin ;
  */
 class ErrorFlow : public Processor
 {
-   public:
-	  /* ====================  LIFECYCLE     ======================================= */
-	  virtual Processor*  newProcessor() { return new ErrorFlow; }
+	public:
```
[the remark this answers is not in the document]
Comment (rete) [reply Analysis/ErrorFlow/include/ErrorFlow.h:50]: @yradkhorrami ping

[the remark this answers is not in the document]
Comment (yradkhorrami) [reply Analysis/ErrorFlow/include/ErrorFlow.h:50]: @rete I did a bit cleaning up! is it ok?

[the remark this answers is not in the document]
Comment (rete) [reply Analysis/ErrorFlow/include/ErrorFlow.h:50]: It looks like the same. You still have additional tabs in the diffs

[the remark this answers is not in the document]
Comment (yradkhorrami) [reply Analysis/ErrorFlow/include/ErrorFlow.h]: @rete should be ok now. yes?

[the remark this answers is not in the document]
Comment (rete) [reply Analysis/ErrorFlow/include/ErrorFlow.h]: Look at the diffs. Not yet

[the remark this answers is not in the document]
Comment (yradkhorrami) [reply Analysis/ErrorFlow/include/ErrorFlow.h]: As I see, a lot of white spaces have been cleaned!
[](https://github.com/iLCSoft/MarlinReco/pull/85/commits/7e3df24ced23e5dd635e6277b00b8c2653228908)

[the remark this answers is not in the document]
Comment (yradkhorrami) [reply Analysis/ErrorFlow/include/ErrorFlow.h]: [](https://github.com/iLCSoft/MarlinReco/pull/85/commits/7e3df24ced23e5dd635e6277b00b8c2653228908)

[the remark this answers is not in the document]
Comment (yradkhorrami) [reply Analysis/ErrorFlow/include/ErrorFlow.h]: @rete did you look at my latest commit?

**Review by rete — APPROVED** [review]

**Pull request merged**